examples/sync/src/resolver.rs

```diff
@@ -51,38 +51,21 @@ where
         }
     }
 
-    /// Connect to the server if not already connected.
-    async fn get_connection(&self) -> Result<(), ResolverError> {
-        let mut connection_guard = self.connection.write().await;
-
-        // Check if we already have a connection
-        if connection_guard.is_some() {
-            return Ok(());
-        }
-
-        // Create new connection
-        info!(server_addr = %self.server_addr, "establishing connection");
-        let (sink, stream) = self
-            .context
-            .dial(self.server_addr)
-            .await
-            .map_err(|e| ResolverError::ConnectionError(format!("Failed to connect: {e}")))?;
-
-        *connection_guard = Some(Connection { sink, stream });
-        info!(server_addr = %self.server_addr, "connected");
-
-        Ok(())
-    }
-
     /// Send a request and receive a response using the persistent connection.
     async fn send_request(&self, request: Message) -> Result<Message, ResolverError> {
```

Comment:
This locking approach prevents us from actually making concurrent requests, right?